entity-service/src/main/kotlin/com/egm/stellio/entity/repository/QueryUtils.kt

```diff
@@ -201,4 +225,24 @@ object QueryUtils {
                }
             """.trimIndent()
         }
+
+    private fun buildGeoQueryFilter(geoQuery: GeoQuery, prefix: String = "AND"): String {
+        val coordinates = geoQuery.coordinates.toString().split("[\\[,\\]]".toRegex())
+        val georelParams = extractGeorelParams(geoQuery.georel!!)
+        return """
+                $prefix entity.location CONTAINS '${geoQuery.geometry!!.uppercase()}'
+                AND point.distance(
+                     point({
+                        latitude: toFloat(apoc.text.regexGroups(entity.location, '([\\d\\.]+) ([\\d\\.]+)')[0][2]), 
+                        longitude: toFloat(apoc.text.regexGroups(entity.location, '([\\d\\.]+) ([\\d\\.]+)')[0][1]),
+                        crs: 'wgs-84'
+                     }),
+                     point({
+                        latitude: ${coordinates[1].toFloat()}, 
+                        longitude: ${coordinates[2].toFloat()}, 
```

Comment:
sure for the lat / lon order??

Reply:
This part corresponds to the coordinates given by the user during the request. 
In this exemple request `http:///ngsi-ld/v1/entities?type=Test1&geoproperty=location&georel=near;maxDistance==1500&geometry=Point&coordinates=[57.5522, -20.3484]` the lat are 57.5522 and the lon are -20.3484 ?

Reply:
don't guess, read the spec. from 4.10 : `A coordinates (parameter name coordinates) element which shall represent the coordinates of the reference geometry as mandated by IETF RFC 7946 [8], section 3.1.1.`